File: Cpp/GPU.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class GPU : public Komponen {
private:
    int vram;  // GB
    int cudaCores;
    string memoryType;  // GDDR6, etc

public:
    GPU() : Komponen() {
        this->vram = 0;
        this->cudaCores = 0;
        this->memoryType = "";
    }

    GPU(int vram, int cudaCores, string memoryType, string merk, string nama, int harga) 
        : Komponen(merk, nama, harga) {
        this->vram = vram;
        this->cudaCores = cudaCores;
        this->memoryType = memoryType;
    }

    void setVram(int vram) {
        this->vram = vram;
    }

    void setCudaCores(int cudaCores) {
        this->cudaCores = cudaCores;
    }

    void setMemoryType(string memoryType) {
        this->memoryType = memoryType;
    }

    int getVram() const {
        return this->vram;
    }

    int getCudaCores() const {
        return this->cudaCores;
    }

    string getMemoryType() const {
        return this->memoryType;
    }

    void displayInfo() const override {
        Komponen::displayInfo();
        cout << "VRAM: " << this->vram << " GB" << endl;
        cout << "CUDA Cores: " << this->cudaCores << endl;
        cout << "Memory Type: " << this->memoryType << endl;
    }

    ~GPU() {}
};

File: Java/GPU.java
public class GPU extends Komponen {
    private int vram;  // GB
    private int cudaCores;
    private String memoryType; // GDDR6, etc

    public GPU() {
        super();
        this.vram = 0;
        this.cudaCores = 0;
        this.memoryType = "";
    }

    public GPU(int vram, int cudaCores, String memoryType, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.vram = vram;
        this.cudaCores = cudaCores;
        this.memoryType = memoryType;
    }

    public void setVram(int vram) {
        this.vram = vram;
    }

    public void setCudaCores(int cudaCores) {
        this.cudaCores = cudaCores;
    }

    public void setMemoryType(String memoryType) {
        this.memoryType = memoryType;
    }

    public int getVram() {
        return this.vram;
    }

    public int getCudaCores() {
        return this.cudaCores;
    }

    public String getMemoryType() {
        return this.memoryType;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("VRAM: " + this.vram + " GB");
        System.out.println("CUDA Cores: " + this.cudaCores);
        System.out.println("Memory Type: " + this.memoryType);
    }
}
